### PR TITLE
refactor flag passing

### PR DESCRIPTION
**Pull request by danlaine**

### CODE REVIEW

File: k8s/utils.go [hunk 1]
```diff
@@ -29,74 +30,64 @@ func convertKey(key string) string {
...
-		} else {
-			// new var
-			v := corev1.EnvVar{
-				Name:  convertKey(key),
-				Value: val.(string),
-			}
-			env = append(env, v)
-		}
+
+	// For each config key, convert it to the format
```
Comment (holisticode) [inline k8s/utils.go:68]: This comment is at the wrong line

File: network/config.go [hunk 2]
```diff
@@ -36,17 +36,16 @@ type AddrAndBalance struct {
 	Balance uint64
 }
 
-// Backend enumerator type
+// The type of network runner
```
Comment (holisticode) [inline network/config.go:39]: Shouldn't comments of exported items start with the exported name?

File: network/config.go [hunk 3]
```diff
@@ -85,12 +83,18 @@ type Config struct {
...
+	// Config flags may also be passed in a node's config struct
+	// or config file.
+	// The precedence of flags handling is, from highest to lowest:
+	// 1. Flags defined in a node's node.Config
+	// 2. Flags defined in a network's network.Config
+	// 3. Flags defined in a node's config file
+	// For example, if a network.Config has flag W set to X,
+	// and a node within that network has flag W set to Y,
+	// and the node's config file has flag W set to Z,
+	// then the node will be started with flag W set to X.
```
Comment (holisticode) [inline network/config.go:97]: It would be set to `Y` - unless I don't understand line 95?

**Review by holisticode — CHANGES_REQUESTED** [review]
some nits

File: network/config.go (hunk 2 above)
Comment (danlaine) [reply network/config.go]: I'm not super committed to that approach -- sometimes I think it makes the comment sound clunky. In any case, changed this comment to start with `Backend`

File: k8s/utils.go (hunk 1 above)
Comment (danlaine) [reply k8s/utils.go]: Moved

File: network/config.go (hunk 3 above)
Comment (danlaine) [reply network/config.go]: woops! fixed, good catch.

**Review by holisticode — APPROVED** [review]

**Pull request merged**